File: CS/DXApplication1/Form1.cs
using DevExpress.LookAndFeel;
using DevExpress.Skins;
using DevExpress.Utils.Svg;
using DXApplication1.Properties;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace DXApplication1
{
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm
    {
        public Form1() {
            InitializeComponent();
            this.FormClosing += Form1_FormClosing;
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e) {
            SavePalette();
        }

        private void ShowSwatchPicker(Form owner) {
            using(var dialog = new DevExpress.Customization.SvgSkinPaletteSelector(owner)) {
                dialog.ShowDialog();
                SavePalette();
            }
        }

        private void SavePalette() {
            var settings = Properties.Settings.Default;
            settings.SkinName = UserLookAndFeel.Default.SkinName;
            settings.Palette = UserLookAndFeel.Default.ActiveSvgPaletteName;
            settings.CompactMode = UserLookAndFeel.Default.CompactUIModeForced;
            settings.Save();
        }

        protected override void OnShown(EventArgs e) {
            base.OnShown(e);
            RestorePalette();
            ShowSettings();
        }

        private void RestorePalette() {
            var settings = Properties.Settings.Default;
            if (!string.IsNullOrEmpty(settings.SkinName)) {
                if (settings.CompactMode)
                    UserLookAndFeel.ForceCompactUIMode(true, false);
                if (!string.IsNullOrEmpty(settings.Palette))
                    UserLookAndFeel.Default.SetSkinStyle(settings.SkinName, settings.Palette);
                else UserLookAndFeel.Default.SetSkinStyle(settings.SkinName);
            }
        }

        private void ShowSettings() {
            labelControl1.Text = Settings.Default.SkinName;
            labelControl4.Text = Settings.Default.Palette != String.Empty ? Settings.Default.Palette : "n/a (default palette or raster skin)";
            labelControl6.Text = Settings.Default.CompactMode ? "Yes" : "No";
        }
    }
}

File: VB/DXApplication1/Form1.vb
Imports DevExpress.LookAndFeel
Imports DXApplication1.Properties
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace DXApplication1

    Public Partial Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Public Sub New()
            InitializeComponent()
            AddHandler FormClosing, AddressOf Form1_FormClosing
        End Sub

        Private Sub Form1_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs)
            SavePalette()
        End Sub

        Private Sub ShowSwatchPicker(ByVal owner As Form)
            Using dialog = New DevExpress.Customization.SvgSkinPaletteSelector(owner)
                dialog.ShowDialog()
                SavePalette()
            End Using
        End Sub

        Private Sub SavePalette()
            Dim settings = Properties.Settings.Default
            settings.SkinName = UserLookAndFeel.Default.SkinName
            settings.Palette = UserLookAndFeel.Default.ActiveSvgPaletteName
            settings.CompactMode = UserLookAndFeel.Default.CompactUIModeForced
            settings.Save()
        End Sub

        Protected Overrides Sub OnShown(ByVal e As EventArgs)
            MyBase.OnShown(e)
            RestorePalette()
            ShowSettings()
        End Sub

        Private Sub RestorePalette()
            Dim settings = Properties.Settings.Default
            If Not String.IsNullOrEmpty(settings.SkinName) Then
                If settings.CompactMode Then UserLookAndFeel.ForceCompactUIMode(True, False)
                If Not String.IsNullOrEmpty(settings.Palette) Then
                    UserLookAndFeel.Default.SetSkinStyle(settings.SkinName, settings.Palette)
                Else
                    UserLookAndFeel.Default.SetSkinStyle(settings.SkinName)
                End If
            End If
        End Sub

        Private Sub ShowSettings()
            labelControl1.Text = Settings.Default.SkinName
            labelControl4.Text = If(Not Equals(Settings.Default.Palette, String.Empty), Settings.Default.Palette, "n/a (default palette or raster skin)")
            labelControl6.Text = If(Settings.Default.CompactMode, "Yes", "No")
        End Sub
    End Class
End Namespace
